c/transform.c

```diff
@@ -94,35 +95,22 @@ void gcj2wgs(double gcjLat, double gcjLng, double *wgsLat, double *wgsLng) {
 }
 
 void gcj2wgs_exact(double gcjLat, double gcjLng, double *wgsLat, double *wgsLng) {
-	double initDelta = 0.01;
-	double threshold = 0.000001;
-	double dLat, dLng, mLat, mLng, pLat, pLng;
-	dLat = dLng = initDelta;
-	mLat = gcjLat - dLat;
-	mLng = gcjLng - dLng;
-	pLat = gcjLat + dLat;
-	pLng = gcjLng + dLng;
+	double dLat, dLng;
+	// n_iter=2: centimeter precision, n_iter=5: double precision
+	const int n_iter = 2;
 	int i;
-	for (i=0; i<30; i++) {
-		*wgsLat = (mLat+pLat) / 2;
-		*wgsLng = (mLng+pLng) / 2;
-		double tmpLat, tmpLng;
-		wgs2gcj(*wgsLat, *wgsLng, &tmpLat, &tmpLng);
-		dLat = tmpLat - gcjLat;
-		dLng = tmpLng - gcjLng;
-		if ( (fabs(dLat)<threshold) && (fabs(dLng)<threshold) ) {
-			return;
-		}
-		if (dLat > 0) {
-			pLat = *wgsLat;
-		} else {
-			mLat = *wgsLat;
-		}
-		if (dLng > 0) {
-			pLng = *wgsLng;
-		} else {
-			mLng = *wgsLng;
-		}
+	if ((wgsLat == NULL) || (wgsLng == NULL)) {
+		return;
+	}
+	*wgsLat = gcjLat;
+	*wgsLng = gcjLng;
+	if (outOfChina(gcjLat, gcjLng)) {
+		return;
+	}
+	for (i = 0; i < n_iter; i++) {
+		delta(*wgsLat, *wgsLng, &dLat, &dLng);
+		*wgsLat = gcjLat - dLat;
+		*wgsLng = gcjLng - dLng;
 	}
```

Comment:
The I'm glad to see this method changed as it's so much faster.

Reply:
Side note... It would be really nice to see consistency between versions and variable names.  X_Pi,xPi,yPi,MPi.....
and consistent definitions across implementations  mPi = 3000*Pi/180;  xPi = x*mPi  for example..  
exact transform implemented via cajun method everywhere...   or left as half distance method, and mars2wgs as cajin.  Just so long as it's all consistent.

Reply:
The C impl is actually the first instance where a caijun-esque method is introduced to eviltransform. Like I said in the PR, it's ditched in an "optimize" commit, the only productive part of which is fixing a "fabs" declaraction.

Reply:
And yeah the whole varname thing is crazy. We probably don't even need all those, as many of them like `x*pi` and `sqrt(fabs(x))` are really subexpression optimizations that can be done automatically by the compiler as long as you allow it to do associative math on floats.

Reply:
I like defining subexpressions  or constants that otherwise would be recalculated multiple times later , to limit the clock cycles needed for repetitive calculations.   I just like to see consistent naming.